hnn_core/tests/test_feed.py

```diff
@@ -97,3 +96,28 @@ def test_extfeed():
     with pytest.raises(ValueError, match='events_per_cycle should be'):
         _create_common_input(distribution, t0, t0_stdev, tstop, f_input,
                              stdev, repeats, events_per_cycle, prng, prng2)
+
+    # test tonic inputs
+    params.update({
+        'N_pyr_x': 3,
+        'N_pyr_y': 3,
+        'tstop': 25,
+        't_evprox_1': 5,
+        't_evdist_1': 10,
+        't_evprox_2': 20,
+        'N_trials': 1,
+        # tonic inputs
+        'Itonic_A_L2Pyr_soma': 1.0,
+        'Itonic_t0_L2Pyr_soma': 5.0,
+        'Itonic_T_L2Pyr_soma': -1.0
+    })
+    net = hnn_core.Network(params)
+
+    with pytest.raises(ValueError, match='End time of tonic input cannot be'
+                       ' negative'):
+        hnn_core.simulate_dipole(net)
+    params['Itonic_T_L2Pyr_soma'] = 4
+    net = hnn_core.Network(params)
+    with pytest.raises(ValueError, match='Duration of tonic input cannot be'
+                       ' negative'):
+        hnn_core.simulate_dipole(net)
```

Comment:
I'd really like to see these tested in `Network`, when the `params` are read from file and set in `net`. So that all (most of) sanity checking is done prior to parallelisation.

Reply:
agreed, will do!